terraform/modules/sns/main.tf

```diff
@@ -64,11 +64,34 @@ data "aws_iam_policy_document" "kms_codestar_access" {
   }
 }
 
+data "aws_iam_policy_document" "kms_eventbridge_access" {
+  statement {
+    sid = "EventBridge SNS KMS Access"
+    actions = [
+      "kms:GenerateDataKey",
+      "kms:Decrypt"
+    ]
+    effect = "Allow"
+    principals {
+      type        = "Service"
+      identifiers = ["events.amazonaws.com"]
+    }
+    resources = ["*"]
+    condition {
+      test     = "StringEquals"
+      variable = "aws:ResourceAccount"
+
+      values = [var.account_id]
+    }
```

Comment:
I'm not entirely sure on this bit as I couldn't find docs, so copied from the S3 one